src/Framework/test/PackageTests.cs

```diff
@@ -0,0 +1,143 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO.Compression;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Testing;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore;
+
+public class PackageTests
+{
+    private readonly string _packageLayoutRoot;
+    private readonly ITestOutputHelper _output;
+
+    public PackageTests(ITestOutputHelper output)
+    {
+        // In official builds, only run this test on Helix. The test will work locally so long as we're building w/ shipping versions.
+        if (!TestData.VerifyPackageAssemblyVersions() && !SkipOnHelixAttribute.OnHelix())
+        {
+            return;
+        }
+
+        _output = output;
+        _packageLayoutRoot = SkipOnHelixAttribute.OnHelix() ?
+            Path.Combine(
+                Environment.GetEnvironmentVariable("DOTNET_ROOT"),
```

Comment:
Again `$(DOTNET_ROOT)` can be located on a readonly file system
```suggestion
                Environment.GetEnvironmentVariable("HELIX_WORKITEM_ROOT"),
```